src/System.Buffers/src/System/Buffers/DefaultArrayPoolBucket.cs

```diff
@@ -45,12 +49,26 @@ internal T[] Rent()
             {
                 _lock.Enter(ref taken);
 
+                var log = ArrayPoolEventSource.Log;
```

Comment:
While it's not terribly expensive, there is a cost to accessing a static like this, and in the (hopefully common) case where a buffer is available, we won't end up logging anything, so this access would be unnecessary.  I'd suggest moving it in to the nested blocks where it's used (I realize that'll mean you'll need to duplicate the access... that's ok).